### ISSUE TITLE
Inconsistent path transformation

### ISSUE THREAD

**Issue by rangp** [vtfuture/RequireJSDotNet#45]
Hello again,
I'm currently exploring your solution and try to integrate it with TypeScripts external module concept.

I have noticed, that you try to maintain the MVC conventions by using capitalized folder and file names and then tranform it to uncapitalized paths to match the require convention.

So the issue I am currently experiencing with this is a bit complicated, by I try to explain it to you.
TypeScript is checking dependencies for modules by the actual file paths, so requiring a module like this works

	import index = require(“Controllers/Root/Home/Index”);

but writing it as the follows results in a compiler-error

	import index = require(“controllers/root/home/Index”);

because TypeScripts dependency-check is case sensitive, just like it is in require.js. 
Now when bundling the files via autoBundles and using overrides I would generate for example a bundle with the following files:

•	 Controllers/Root/Home/Index
•	Controllers/Root/Home/Extended

And the corresponding generated path overrides would look like this:

    {
     "controllers/root/home/Index": "bundles/baseBundle",
     "controllers/root/home/Extended": "bundles/baseBundle"
    }

Now my baseBundle file looks looks like follows with code compiled by TypeScript

       define(“controllers/root/home/Index”, [“module”,”exports”], function(){..});
       define(“controllers/root/home/Extended”,
                  [“module”,”exports”, “ Controllers/Root/Home/Index”], function(){..});

And there things go down. The module “controllers/root/home/Index” is properly defined, but require.js sees the dependency “ Controllers/Root/Home/Index” as a different file and loads it again,
ending in having loaded the file-contents twice.

In my opinion this is a rather inconsistent behavior,
because I would expect the compressor either to de-capitalize all of the dependencies or none of them.

Maybe there should be the possibility to disable de-capitalization of file paths as an option, for I do not see the need of it, 
except matching  the conventions?

And by the way I am very impressed by this projects.
I see it as a great solution to bring AMD concepts into multipage apps.

Cheers!

**Comment by CezarCretu** [comment]
While I agree that such inconsistencies are more trouble than they're worth in the long run, I'm reluctant to change this right now since the implications would be far reaching for existing projects. That being said, it's something we should fix for 3.0.

As a band-aid for your case I've made the autobundler apply the same transformation to require() dependencies that it applies to file paths when generating a module name, so in your case baseBundle should look like this:

```
 define(“controllers/root/home/Index”, [“module”,”exports”], function(){..});
   define(“controllers/root/home/Extended”,
              [“module”,”exports”, “ controllers/root/home/Index”], function(){..});
```

Please update both the require and require compressor packages to 2.1 and tell me if that works for you.

**Comment by rangp** [comment]
I tried and it works fine now for me!
Thank you very much.
